### ISSUE TITLE
Condider making first argument of input() mandatory

### ISSUE THREAD

**Issue by jnizet** [angular/angular#53909]
### Which @angular/* package(s) are relevant/related to the feature request?

core

### Description

There is a difference in the design of the `input()` and of the `signal()` functions.

`signal<T>()` expects an argument and returns a `Signal<T>`, whereas `input<T>()` can be called with or without argument and returns a `Signal<T | undefined>` if no argument is passed.

Moreover, if we have code that uses `input()` without argument:

```ts
collapsed = input<boolean>();
```

and want to pass it options, then suddenly we're forced to 
- pass an initial value, although the initial value is the same as before;
- change the generic type passed to the function, even though the type of the signal to create is the same as before:

```ts
collapsed = input<boolean | undefined>(undefined, { alias: 'appCollapsed' });
```

### Proposed solution

Even though it requires a bit more code, I find the the result would be more consistent, more explicit, and less surprising if `input()` always required an initial value:

```ts
collapsed = input<boolean | undefined>(undefined);
```

- adding options requires just that: adding options
- the generic type is the same as the generic type of the created Signal
- it works the same way as signal()

### Alternatives considered

NA

**Comment by JeanMeche** [comment]
@zip-fa About that topic, @jelbourn  explains in friday's stream why the team went with optional by default : https://twitter.com/Jean__Meche/status/1746340393048826088

**Comment by e-oz** [comment]
> adding options requires just that: adding options

The best argument in my opinion

**Comment by simeyla** [comment]
It's just too ugly. Then imagine ten inputs in row. 
I expect to rarely be using options myselfand in this example boolean probably don't need to be tri-state and can just have a default of false.

**Comment by JoostK** [comment]
I don't think that an initial value should be made mandatory, but I can see how an overload like

```ts
function input<ReadT>(initialValue: undefined, options: InputOptionsWithoutTransform<ReadT>): InputSignal<ReadT|undefined>;
```

could potentially help avoid that awkward generic type.

**Comment by devversion** [comment]
We did collect some data internally when considering various APIs. Around 42% of inputs were not providing an explicit value and it showed a convenience shorthand might be desirable to have. Repeating `undefined` all the time would be quite cumbersome.

The fact that the generic does not match with the return type of the `InputSignal` is a bigger concern for me than having to specify an explicit `undefined` when adding options. Most of the time, we don't assume options to be used. 

I like the overload suggested by @JoostK, but in either case (with current assumptions) will be a special rule- and it feels less invasive if that rule only applies to `input()` without any arguments, compared to also expanding the rule to `input(undefined)`.

**Comment by muuvmuuv** [comment]
Why should a component input be required anytime when some components might not need every input to be fullfilled? It is a common case, like a map component that shows a pin if a pin coordinate is given otherwise not:

```ts
readonly pinCoordinates = signal<MapCoordinates | undefined>(undefined)
```

I would go the otherway and say that a initial value should not be required, currently I must explicitly give it `undefined` which is obsolete. IMO it should be e.g.:

```ts
export declare function signal<T>(options?: CreateSignalOptions<T>): WritableSignal<T>;
export declare function signal<T>(initialValue?: T, options?: CreateSignalOptions<T>): WritableSignal<T>;

readonly pinCoordinates = signal<MapCoordinates>() // => MapCoordinates|undefined
readonly pinCoordinates = signal<MapCoordinates>({/*...*/}) // => MapCoordinates
readonly pinCoordinates = signal.required<MapCoordinates>() // => MapCoordinates
```

Or does the Signal need a value for some internal processing/caching?

**Comment by JoostK** [comment]
Closing as the approach in https://github.com/angular/angular/issues/53909#issuecomment-1891089694 has been applied in #57621, landing in v19. This isn't exactly as requested here, but should result in improved developer experience.
